tests/hf_hydrodata/test_point.py

```diff
@@ -1725,5 +1725,67 @@ def test_get_site_variables_jasechko():
     assert len(df) == 281
 
 
+def test_get_data_fan():
+    """Test getting data for Fan dataset."""
+    df = point.get_point_data(
+        dataset="fan_2013",
+        variable="water_table_depth",
+        temporal_resolution="long_term",
+        aggregation="mean",
+        grid="conus2",
+        grid_bounds=[1500, 1300, 1550, 1500],
+    )
+    assert len(df) == 187
+
+
+def test_get_metadata_fan():
+    """Test getting metadata for Fan dataset."""
+    metadata_df = point.get_point_metadata(
+        dataset="fan_2013",
+        variable="water_table_depth",
+        temporal_resolution="long_term",
+        aggregation="mean",
+        grid="conus2",
+        grid_bounds=[1500, 1300, 1550, 1500],
+    )
+    assert len(metadata_df) == 187
+
+
+def test_get_citations_fan():
+    """Test citations for fan_2013 dataset."""
+    t = point._get_point_citations("fan_2013")
+    assert t == "Dataset DOI: 10.1126/science.1229881"
+
+
+def test_get_site_variables_fan():
+    """Test for get_site_variables with Fan dataset."""
+    df = point.get_site_variables(
+        dataset="fan_2013",
+        variable="water_table_depth",
+        date_start="2000-01-01",
+        date_end="2022-12-31",
+        state="TX",
+        latitude_range=(27, 28),
+    )
+    assert len(df) == 22
+
+
+def test_fail_fan_outside_timerange():
+    """
+    Test to ensure proper failure if filtering on dates outside of the
+    date range which the Fan dataset is available for (1927-2009).
+    """
+    with pytest.raises(Exception) as exc:
+        point.get_point_data(
+            dataset="fan_2013",
+            variable="water_table_depth",
+            temporal_resolution="long_term",
+            aggregation="mean",
+            date_start="2010-01-01",
+            date_end="2010-12-31",
```

Comment:
Is there a different expected behavior if the date range overlaps the datasets boundary, that we can/should test? Like a request that goes from `2009/12/01 - 2010/02/01` ?

Reply:
Good idea - I added another test in that uses an overlapping date range. This dataset is a little odd because the data itself does not vary temporally. It represents the long-term mean for each site location so the date range doesn't really change the data that gets returned (the dataset is one observation per site).